### PR TITLE
feat: init gemini llm

### PR DESCRIPTION
**Pull request by thucpn**
Follow logic from llama-index repo:
https://github.com/run-llama/llama_index/blob/main/llama-index-integrations/llms/llama-index-llms-gemini/llama_index/llms/gemini/base.py

### CODE REVIEW

**Review by yisding — APPROVED** [review]
This looks good thanks! Let's also add Gemini-1.5 because I know people will want to experiment with the long context windows even though it's still in beta.

File: packages/core/src/llm/gemini.ts [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+
+/**
+ * Gemini Session to manage the connection to the Gemini API
+ */
+export class GeminiSession {
+  gemini: GoogleGenerativeAI;
+
+  constructor(options: GeminiSessionOptions) {
+    if (!options.apiKey) {
+      options.apiKey = getEnv("ANTHROPIC_API_KEY");
```
Comment (marcusschiesser) [inline packages/core/src/llm/gemini.ts]: Wrong key name?

File: packages/core/src/llm/gemini.ts [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+    const result = await client.generateContent(
+      GeminiHelper.messageContentToGeminiParts(prompt),
+    );
+    return {
+      text: result.response.text(),
+      raw: result.response,
+    };
+  }
+
+  private async getEmbedding(prompt: string): Promise<number[]> {
```
Comment (marcusschiesser) [inline packages/core/src/llm/gemini.ts:368]: Better seperate embedding and LLM model usage in two classes (like the other LLM implementations except Ollama)

File: packages/core/src/llm/gemini.ts [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+      this.sessions.push({ session, options });
+    }
+    return session;
+  }
+}
+
+/**
+ * Helper class providing utility functions for Gemini
+ */
+export class GeminiHelper {
```
Comment (marcusschiesser) [inline packages/core/src/llm/gemini.ts:141]: Needs to be exported?

File: packages/core/src/llm/gemini.ts [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+      throw new Error("Set Google API Key in GOOGLE_API_KEY env variable");
+    }
+    this.gemini = new GoogleGenerativeAI(options.apiKey);
+  }
+}
+
+/**
+ * Gemini Session Store to manage the current Gemini sessions
+ */
+export class GeminiSessionStore {
```
Comment (marcusschiesser) [inline packages/core/src/llm/gemini.ts:113]: Do we need multiple sessions?

**Review by marcusschiesser — COMMENTED** [review]
@thucpn please add also to docs with a short example

**Review by marcusschiesser — APPROVED** [review]
Thanks @thucpn that looks great!

**Pull request merged**